### PR TITLE
Split up `decode_test` into tests focused on individual SQL protos.

### PR DESCRIPTION
**Pull request by markww**
In PR #421, @bgogul noted that `decode_test.cc` was rather long and had
little code sharing between its tests and asked that it be split up to
improve readability. This PR does that.

### CODE REVIEW

**Review by bgogul — COMMENTED** [review]
Test seems to be faling?

File: src/frontends/sql/decode_merge_operation_test.cc [hunk 1]
```diff
@@ -0,0 +1,137 @@
...
+#include "google/protobuf/text_format.h"
+#include "google/protobuf/util/message_differencer.h"
+#include "src/common/testing/gtest.h"
+#include "src/frontends/sql/decode.h"
+#include "src/frontends/sql/decoder_context.h"
+#include "src/frontends/sql/sql_ir.pb.h"
+#include "src/frontends/sql/testing/merge_operation_view.h"
+
+namespace raksha::frontends::sql {
+
```
[the remark this answers is not in the document]
Comment (markww) [reply src/frontends/sql/decode_merge_operation_test.cc:27]: Fixed.

**Comment by markww** [comment]
Got rid of the unused `constexpr` global causing the failure, hopefully tests should pass now.

**Comment by markww** [comment]
Alright, tests are passing now and I fixed the namespace issue, this should be ready for another look.

**Review by bgogul — APPROVED** [review]

**Comment by bgogul** [comment]
The error during import is as follows:

> 0308 12:45:33.873 ERROR: Buildozer could not find a target for src/frontends/sql:decode_test: rule 'decode_test' not found. Use --ignore-noop if you want to ignore this error

It looks like we need to fix some copybara transformations?

**Comment by markww** [comment]
Yes, that's correct. Let me see what I can do here.

**Pull request closed without merging**